PusherClient/Pusher.cs

```diff
@@ -100,24 +101,14 @@ public Pusher(string applicationKey, PusherOptions options = null)
 
         #region Public Methods
 
+        [MethodImpl(MethodImplOptions.Synchronized)]

```

Comment:
Instead of using this, I'd recommend we use "lock(myVar)" instead.  For example:

a private member:

private readonly object _lockingObject = new object();

then wrap the contents of the Connect method in:

lock(_lockingObject)
{
    // existing code
}

In addition, I think we'd want to lock on Disconnect to.  I'll have a look in more detail this evening.  I'm also questioning why we want to recreate the Connection object everytime we connect, more investigation needed.